tdrs-frontend/src/components/ComboBox/ComboBox.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useRef } from 'react'
+import PropTypes from 'prop-types'
+import comboBox from 'uswds/src/js/components/combo-box'
+
+const ComboBox = ({ sttList }) => {
+  const selectRef = useRef()
+  useEffect(() => {
+    // The combo box was not rendering as a combo box without this line
+    comboBox.init()
+    // This solved the issue when tabbing through the form on EditProfile,
+    // a selection was automatically being made on the first option
+    selectRef.current.value = ''
+  })
+  return (
+    <label className="usa-label" htmlFor="sttList">
+      Associated State, Tribe, or Territory
+      <div className="usa-combo-box">
+        <select className="usa-select" name="stt" id="stt" ref={selectRef}>
+          {sttList.map((stt) => (
+            <option
+              className="sttOption"
+              key={stt.id}
+              value={stt.name.toLowerCase()}
+            >
+              {stt.name}
+            </option>
+          ))}
+        </select>
+      </div>
+    </label>
+  )
+}
+
+ComboBox.propTypes = {
+  sttList: PropTypes.arrayOf(
+    PropTypes.shape({
+      name: PropTypes.string,
+      id: PropTypes.number,
+    })
+  ).isRequired,
+}
+
+export default ComboBox
```

Comment:
I pulled out the ComboBox component to clean up EditProfile which was becoming really long and complex. I am happy to add it back into the EditProfile if anyone thinks that it is too cumbersome to have it separated.

Reply:
I think it's much better to have it pulled into its own component. The one change I'd like to see is to generalize it a bit more so it can be reused elsewhere without passing options in using `sttList`.

Reply:
I totally agree. I made the change to just pass in `children` from the parent component and that removes some of the stuff I have to pass in later on any way so double win! I'm happy to include that in the next PR for 230 - Form Validation but I pushed it to the 229 branch if you want to see what I implemented before then. [Combobox portion](https://github.com/raft-tech/TANF-app/blob/229-combobox/tdrs-frontend/src/components/ComboBox/ComboBox.jsx#L5)

[EditProfile](https://github.com/raft-tech/TANF-app/blob/229-combobox/tdrs-frontend/src/components/EditProfile/EditProfile.jsx#L48-L56)

Reply:
I'm totally comfortable with tabling the changes to generalize this class to the next PR.

Just as early feedback on those changes, the `<select>` component still is referencing `stt` in the `name` and `id` fields, as well as the `<label htmlFor=>`. I think generalized defaults, with the ability to override them in the `ComboBox` props would be nice.

Reply:
Also as early feedback, I like using `{children}` to deal with passing the options in. Good work.